src/base/detail/RangeImpl.hh

```diff
@@ -44,6 +44,12 @@ struct RangeTypeTraits
         v += i;
         return v;
     }
+    static CELER_FORCEINLINE_FUNCTION value_type decrement(value_type   v,
+                                                           counter_type i)
+    {
+        v -= i;
+        return v;
```

Comment:
What is the advantage of the code use over 'just'
```
   return v - i;
```
since 'v' does not appear to be a reference?

Reply:
Requires defining an extra operator for the detail iterator